### ISSUE TITLE
"Delete and Skip Media" adds the video to download queue

### ISSUE THREAD

**Issue by zpz5HAU-tgc3fgw2xwr** [meeb/tubesync#280]
As the title says, hitting the "Delete and Skip Media" button keeps re-queueing the video in the download queue. This doesn't happen with every video, but when it does happen it's consistent.

It's possible this is somehow related to me initially deleting the video file from Plex. But even after the video re-downloads if I go in and tell TubeSync to "Delete and Skip Media" instead of manually deleting, it still requeues.

I'm mainly trying to get rid of some old entries in TubeSync on the media tab. Files that I either deleted manually, or should have been deleted due to TubeSync auto delete rules. Maybe something that could help with this is the ability to rescan local media. If TubeSync thinks a local file exists, scan for it, delete it if it's missing. Something like this could also help if you decide to add a renaming feature... say you change your name rules for a channel, rescan local media, and apply the new name rules to the file. A re-scan/renaming feature would be very helpful in cases where you've archived an entire channel.

**Comment by meeb** [comment]
Yes there are a couple of situations where the skip flag behaves weirdly. It was originally just an internal flag that sets media to be skipped if it doesn't match the download criteria but this was adapted to be manually set-able. Ideally I need to add a second skip flag, one for manual skips and one for internal parameter mismatch skips.

It is likely deleting something may tweak a skip flag as that's when "can media be downloaded?" is calculated from the source requirements vs. media metadata (especially if the metadata has been updated since the last check).

A rename feature is on the wishlist i believe.

**Comment by tcely** [comment]
I believe everything mentioned in this issue is now available.

@meeb Please close this when you have a moment.
